data_model/src/isi.rs

```diff
@@ -94,138 +83,175 @@ impl Instruction {
-    /// Key string.
-    pub key: EvaluatesTo<Name>,
-}
-
-/// Sized structure for all possible Registers.
-#[derive(
-    Debug, Display, Clone, PartialEq, Eq, Decode, Encode, Deserialize, Serialize, IntoSchema, Hash,
-)]
-#[display(fmt = "REGISTER `{object}`")]
-#[serde(transparent)]
-#[repr(transparent)]
-pub struct RegisterBox {
-    /// The object that should be registered, should be uniquely identifiable by its id.
-    pub object: EvaluatesTo<RegistrableBox>,
-}
-
-/// Sized structure for all possible Unregisters.
-#[derive(
-    Debug, Display, Clone, PartialEq, Eq, Decode, Encode, Deserialize, Serialize, IntoSchema, Hash,
-)]
-#[display(fmt = "UNREGISTER `{object_id}`")]
-#[serde(transparent)]
-#[repr(transparent)]
-pub struct UnregisterBox {
-    /// The id of the object that should be unregistered.
-    pub object_id: EvaluatesTo<IdBox>,
-}
-
-/// Sized structure for all possible Mints.
-#[derive(
-    Debug, Display, Clone, PartialEq, Eq, Decode, Encode, Deserialize, Serialize, IntoSchema, Hash,
-)]
-#[display(fmt = "MINT `{object}` TO `{destination_id}`")]
-pub struct MintBox {
-    /// Object to mint.
-    #[serde(flatten)]
-    pub object: EvaluatesTo<Value>,
-    /// Entity to mint to.
-    pub destination_id: EvaluatesTo<IdBox>,
-}
-
-/// Sized structure for all possible Burns.
-#[derive(
-    Debug, Display, Clone, PartialEq, Eq, Decode, Encode, Deserialize, Serialize, IntoSchema, Hash,
-)]
-#[display(fmt = "BURN `{object}` FROM `{destination_id}`")]
-pub struct BurnBox {
-    /// Object to burn.
-    #[serde(flatten)]
-    pub object: EvaluatesTo<Value>,
-    /// Entity to burn from.
-    pub destination_id: EvaluatesTo<IdBox>,
-}
-
-/// Sized structure for all possible Transfers.
-#[derive(
-    Debug, Display, Clone, PartialEq, Eq, Decode, Encode, Deserialize, Serialize, IntoSchema, Hash,
-)]
-#[display(fmt = "TRANSFER `{object}` FROM `{source_id}` TO `{destination_id}`")]
-pub struct TransferBox {
-    /// Entity to transfer from.
-    pub source_id: EvaluatesTo<IdBox>,
-    /// Object to transfer.
-    #[serde(flatten)]
-    pub object: EvaluatesTo<Value>,
-    /// Entity to transfer to.
-    pub destination_id: EvaluatesTo<IdBox>,
-}
-
-/// Composite instruction for a pair of instructions.
-#[derive(
-    Debug, Display, Clone, PartialEq, Eq, Decode, Encode, Deserialize, Serialize, IntoSchema, Hash,
-)]
-#[display(fmt = "(`{left_instruction}`, `{right_instruction}`)")]
-pub struct Pair {
-    /// Left instruction
-    pub left_instruction: Instruction,
-    /// Right instruction
-    pub right_instruction: Instruction,
-}
-
-/// Composite instruction for a sequence of instructions.
-#[derive(Debug, Clone, PartialEq, Eq, Decode, Encode, Deserialize, Serialize, IntoSchema, Hash)]
-pub struct SequenceBox {
-    /// Sequence of Iroha Special Instructions to execute.
-    pub instructions: Vec<Instruction>,
+macro_rules! isi {
+    ($($meta:meta)* $item:item) => {
+        crate::model! {
```

Comment:
`Derive(Display)` as well? I think the exception (`SequenceBox`) could just have the macro inlined.

Reply:
I would have done that but `Display` wasn't playing nicely with my macro for some reason which I was not able to determine so I just left it out. I'll try one more time

Reply:
this pops up: `cannot find value `_derive_more_display_formatter` in this scope`. I'm leaving it as is